android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateUtils.java

```diff
@@ -176,11 +180,14 @@ public static void verifyFolderHash(String folderPath, String expectedHash) {
 
     public static Map<String, Object> verifyAndDecodeJWT(String jwt, PublicKey publicKey) {
         try {
-            final JWTVerifier verifier = new JWTVerifier(publicKey);
-            final Map<String, Object> claims = verifier.verify(jwt);
-            CodePushUtils.log("JWT verification succeeded:\n" + claims.toString());
-            return claims;
-        } catch (Exception e) {
+            SignedJWT signedJWT = SignedJWT.parse(jwt);
+            JWSVerifier verifier = new RSASSAVerifier((RSAPublicKey)publicKey);
+            if (signedJWT.verify(verifier)) {
+                CodePushUtils.log("JWT verification succeeded:\n" + claims.toString());
+                return signedJWT.getJWTClaimsSet().getClaims();
+            }
+            return null;
+        } catch (Exception ex) {
             return null;
```

Comment:
I think we need to add exception logging here to be able determine cause of problem in future:
```java
CodePushUtils.log(e.getMessage());
CodePushUtils.log(e.getStackTrace().toString());
```